dashboard/app/controllers/pd/application/teacher_application_controller.rb

```diff
@@ -16,12 +16,18 @@ def new
       return render :not_teacher unless current_user.teacher?
       return render :no_teacher_email unless current_user.email.present?
 
+      @year = APPLICATION_CURRENT_YEAR
+
       @application = TEACHER_APPLICATION_CLASS.
-        where(application_year: APPLICATION_CURRENT_YEAR).
+        where(application_year: @year).
```

Comment:
we should probably make a note somewhere that during the archiving process, teachers will lose access to their old (incomplete) applications, or handle an incomplete application differently when considering what year it is...

Reply:
i.e. when loading an incomplete application, grab the most recent one? assuming teachers can only have one incomplete application going at a time. (we might also want to update the year of the application if they're filling it out after coming back from a year)

Reply:
Hmm, good call ... I was thinking to grab their application from the "current" year, not caring if there's a recent one from a different year. If they only have one from a different year, don't use any of that data?

Reply:
yeah, in that case, they'll lose that data, which could be as recent as a month old (they started an application just before they closed), so we'd probably want to communicate that somehow somewhere?

Reply:
Hmm, good call. I'll ask Tess about it

Reply:
UPDATE: Checked with Tess on if an application was found for a previous year. If so, add a message saying, "We see you started an application for the [last year] cycle, we are now accepting applications for [this year] school year.  Please check out the new application and apply! [if they have a partner] If you have any questions please reach out to your regional partner, {​regional partner name} at {​regional partner email} [if they dont have a partner] Your application progress has been saved. If you have any questions please reach out to Code.org at support@code.org"